src/components/FeatureViews/AdminPortal/ActionLog.js

```diff
@@ -113,57 +126,41 @@ const ActionLog = () => {
             sort: currentSort.sort,
             objectType: objectType.toLowerCase(),
             userId: currentId.toString(),
-        },
-        "onCompleted": (data) => {
-            setLogData(data);
-        },
+        }
     });
     const [userOptions, setUserOptions] = useState(null)
     const adminOptions = ["Owner", "Receptionist", "Assistant"];
     const actionOptions = ["Add", "Edit", "Delete"]
     const objectOptions = ["Student", "Admin", "Parent", "Instructor", "Payment", "Registration", "Tutoring", "Course", "Discount", "Price Rules"]
 
-    useEffect(() => {
-        logs()
-        fetchAdmins()
-    }, [])
-
-    const renderChip = (action) => {
-        switch (action) {
-            case "Edit":
-                return (<Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#FFDD59" }} label="Edit"></Chip>)
-            case "Add":
-                return (<Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#78E08F" }} label="Add"></Chip>)
-            case "Delete":
-                return (<Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#FF7675" }} label="Delete"></Chip>)
-        }
-    }
+    if (loading || admins.loading) return <Loading />;
+    // if (error || admins.error) return "error";
+    // for some reason returning error will break on filtering for objects. Not sure why, maybe it's a backend issue
 
-    const handleAdminSelection = (event) => {
-        const value = event.target.value;
-        setAdminType(value);
-    }
+    console.log(data);
+    console.log(objectType);
+    const renderChip = (action) => ({
+        "Edit": <Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#FFDD59" }} label="Edit" />,
+        "Add": <Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#78E08F" }} label="Add" />,
+        "Delete": <Chip style={{ borderRadius: 3, width: 70, backgroundColor: "#FF7675" }} label="Delete" />,
+    }[action]);
 
-    const handleObjectSelection = (event) => {
+    const handleSelection = (setValue) => (event) => {
         const value = event.target.value;
-        setObjectType(value);
+        console.log(value);
+        setValue(value);
     }
 
     const handleUserSelection = (event) => {
         setUserType(event.target.value)
-        userOptions.map((userType) => {
+        admins.data.admins.map((userType) => {
             if (userType === event.target.value) {
                 setCurrentId(userType.user.id);
```

Comment:
rewrite this to use the .filter() function instead of .map(). also, there could potentially be multiple currentId's if there are multiple users with the same userTypes, is this what you intended? or are you just trying to find the first match?